src/lib/operators/join_nested_loop_a.cpp

```diff
@@ -327,11 +366,14 @@ class JoinNestedLoopA::JoinNestedLoopAImpl : public AbstractJoinOperatorImpl {
     } else {
       for (ChunkOffset row_left = 0; row_left < size_left; ++row_left) {
         for (ChunkOffset row_right = 0; row_right < size_right; ++row_right) {
-          auto is_match = context->compare_func(get_left_column_value(row_left), get_right_column_value(row_right));
-
           auto current_right = RowID{context->chunk_id_right, row_right};
           auto current_left = RowID{context->chunk_id_left, row_left};
-          if (is_match) {
+
+          auto is_match = context->compare_func(get_left_column_value(row_left), get_right_column_value(row_right));
+
+          auto comparing_null = is_left_value_null(row_left) || is_right_value_null(row_right);
+
+          if (!comparing_null && is_match) {
```

Comment:
same as above. when comparing with `NULL`, there is no need to call `compare_func` (I think)